coala_langserver/langserver.py

```diff
@@ -110,7 +102,37 @@ def send_diagnostics(self, path, diagnostics):
             'uri': 'file://{0}'.format(path),
             'diagnostics': _diagnostics,
         }
-        self.send_notification('textDocument/publishDiagnostics', params)
+        self._endpoint.notify('textDocument/publishDiagnostics', params=params)
+
+
+@enforce_signature
+def start_tcp_lang_server(handler_class: LangServer, bind_addr, port):
+    # Construct a custom wrapper class around the user's handler_class
+    wrapper_class = type(
+        handler_class.__name__ + 'Handler',
+        (_StreamHandlerWrapper,),
+        {'DELEGATE_CLASS': handler_class}
+    )
+
+    try:
+        server = socketserver.TCPServer((bind_addr, port), wrapper_class)
+    except Exception as e:
+        log('Fatal Exception: {}'.format(e))
+        sys.exit(1)
+    try:
+        log('Serving {} on ({}, {})'.format(
```

Comment:
`log` isnt useful inside the try.

but, it is a new `log` call, and redundant due to existing `Accepting TCP connections ..` log

Reply:
I have to retain this log compared to the other because the tests watch the log calls to trigger sending requests.